### PR TITLE
Add dataflow runner to scio-examples

### PR DESCRIPTION
**Pull request by regadas**
I noticed that almost all examples reference `--runner=DataflowRunner`. Not sure if we just add the dep or update them to use `DirectRunner`

### CODE REVIEW

**Comment by nevillelyh** [comment]
I'm a little conflicted. Kinda want to remove runner deps for the sake of cleanness, but also find myself adding it back often to test a job. I'd prefer if there's a way to add runner deps via some configuration flag, like a maven profile. Any idea how to do so in sbt?

**Comment by regadas** [comment]
@nevillelyh something like this?

```
sbt scio-examples/dataflow:runMain ...

or
set scio-examples/direct:runMain ...
```

**Comment by nevillelyh** [comment]
That'll require custom sbt tasks though? I'm thinking more like `-Drunner=...` to modify dependencies, but that's not perfect either. Open to other ideas :)

**Comment by regadas** [comment]
@nevillelyh as is no need for custom tasks by prefixing the default tasks you are only adding the respective runner dep to the classpath.

**Comment by nevillelyh** [comment]
Cool that's close to what I was thinking.

**Comment by nevillelyh** [comment]
LGTM, how about making direct the default though, so that `sbt runMain WordCount ...` still works out of the box? We should update examples & wiki also once this is in.

**Comment by regadas** [comment]
@nevillelyh sounds good!

File: build.sbt [hunk 1]
```diff
@@ -506,13 +514,15 @@ lazy val scioSchemas: Project = Project(
   compileOrder := CompileOrder.JavaThenScala
 )
 
+
```
Comment (nevillelyh) [inline build.sbt]: Nitpicking: extra blank line

**Comment by regadas** [comment]
@nevillelyh decided to fallback to the idea of having `-DbeamRunner=` because:
1) using configs and defaults was starting to get complicated.
2) running job requires the `--runner` to be set so it would become weird to have `dataflow:runMain ... --runner=DataflowRunner`

with this new changes you can 

```
// if -DbeamRunners not set default is DirectRunner
sbt -DbeamRunners=DirectRunner,DataflowRunner 
```

Also, there a couple more sub projects that use it. `scioTest` `scioRepl` should we use the same approach?

**Comment by regadas** [comment]
@nevillelyh let me know what you think about this 😄

File: build.sbt [hunk 2]
```diff
@@ -510,9 +532,8 @@ lazy val scioExamples: Project = Project(
   "scio-examples",
   file("scio-examples")
 ).settings(
-  commonSettings ++ noPublishSettings ++ soccoSettings,
+  commonSettings ++ noPublishSettings ++ soccoSettings ++ beamRunnerSettings,
   libraryDependencies ++= Seq(
-    "org.apache.beam" % "beam-runners-direct-java" % beamVersion,
```
Comment (nevillelyh) [inline build.sbt]: We should either keep this or change `def beamRunnerSettings` to include `DirectRunner` by default?

Comment (regadas) [reply build.sbt]: @nevillelyh `beamRunnerSettings` includes `DirectRunner` if no `-DbeamRunners` or `BEAM_RUNNERS=` is found. 

https://github.com/spotify/scio/pull/963/files#diff-fdc3abdfd754eeb24090dbd90aeec2ceR207

https://github.com/spotify/scio/pull/963/files#diff-fdc3abdfd754eeb24090dbd90aeec2ceR209

Comment (regadas) [reply build.sbt]: @nevillelyh the idea is that ... if we need a runner in some subproject we just include beamRunnerSettings

Comment (nevillelyh) [reply build.sbt]: LOL missed `.getOrElse(Seq(directRunnerDependency))`. LGTM.

**Comment by regadas** [comment]
@nevillelyh now you can do 
```
sbt:scio-examples> set beamRunners := "DirectRunner,DataflowRunner"
```

**Pull request merged**